app-modules/integration-twilio/src/Actions/StatusCallback.php

```diff
@@ -55,6 +57,21 @@ public function __construct(
 
     public function handle(): void
     {
-        // TODO Handle a status update event
+        $deliverable = EngagementDeliverable::where('external_reference_id', $this->data['MessageSid'])->first();
+
+        if (is_null($deliverable)) {
+            // TODO Potentially trigger a notification to an admin that a message was received for a non-existent deliverable
+            return;
+        }
+
+        // TODO We should implement some sort of process that checks to see if a deliverable has been updated to the "delivered" or "undelivered"
+        // status after a certain period of time. This is to handle an edge case where the webhook is not received for some reason, and in this
+        // situation we can simply poll Twilio for the data related to this deliverable. It can be a simple process implemented through the Kernel
+
+        // TODO In order to potentially reduce the amount of noise from jobs, we might want to introduce a "screener" that eliminates certain jobs based on their status
+        // And only run the update if it's a status that we want to run some type of update against. For instance, we will receive callbacks for
+        // queued, sending, sent, etc... but we don't actually want/need to do anything during these lifecycle hooks. We only really care about
+        // delivered, undelivered, failed, etc... statuses.
```

Comment:
We might want to do what I did for email event webhooks and create events per the type of webhook it is. And then we can have (or not have) listeners for those specific events.